src/python/pants/backend/python/lint/black/rules.py

```diff
@@ -124,6 +126,8 @@ def create(
 
 @rule(name="Format using Black")
 async def fmt(wrapped_target: BlackTarget, black_setup: BlackSetup) -> FmtResult:
+  if black_setup.skip:
+    return FmtResult.noop()
```

Comment:
This isn't perfect. We will set up Black before deciding to no-op when `--black-skip`. The Black setup is very frequently cached as it's just the Black Pex + passthrough args + config file, so every target in a run will share the same setup. But, still, this is not ideal.

I tried `await Get[BlackSetup]` in the rule body and couldn't get it working. I think we need multiple params support: https://github.com/pantsbuild/pants/issues/7490